src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpExporterOptionsExtensions.cs

```diff
@@ -58,23 +58,33 @@ public static THeaders GetHeaders<THeaders>(this OtlpExporterOptions options, Ac
         {
             // According to the specification, URL-encoded headers must be supported.
             optionHeaders = Uri.UnescapeDataString(optionHeaders);
+            ReadOnlySpan<char> headersSpan = optionHeaders.AsSpan();
 
-            Array.ForEach(
-                optionHeaders.Split(','),
-                (pair) =>
+            while (!headersSpan.IsEmpty)
+            {
+                int commaIndex = headersSpan.IndexOf(',');
+                ReadOnlySpan<char> pair;
+                if (commaIndex == -1)
                 {
-                    // Specify the maximum number of substrings to return to 2
-                    // This treats everything that follows the first `=` in the string as the value to be added for the metadata key
-                    var keyValueData = pair.Split(['='], 2);
-                    if (keyValueData.Length != 2)
-                    {
-                        throw new ArgumentException("Headers provided in an invalid format.");
-                    }
+                    pair = headersSpan;
+                    headersSpan = [];
```

Comment:
```suggestion
                    headersSpan = ReadOnlySpan<char>.Empty;
```

Reply:
Why? The `[]` is the recommended way by the analyzers. IDE0300/IDE0301 as I remember?

Reply:
Thanks for the mention. I didn't know it also covers Span.